src/main/resources/checkstyle-config.xml

```diff
@@ -35,11 +35,26 @@
 
     <module name="AvoidStarImport"/>
     <module name="NeedBraces"/>
+    <module name="EmptyStatement"/>
 
   </module>
 
   <module name="FileTabCharacter"/>
 
+  <!--
+    Due to a bug in openjdk 7 and 8, extra semicolons in import statements are allowed. However, these extra semicolons can cause compilation failures with other java
+    distributions since they are technically disallowed by the java spec. These two regexes prevent extra semicolons in import statements. As an extra side-effect they also
+    disallow certain usages of semicolons elsewhere in the body of a java file. But we are okay with disallowing strange placement of semicolons in those other cases.
+  -->
+  <module name="RegexpSingleline">
```

Comment:
EmptyStatement and OneStatementPerLine don't work for imports so we still need to add a somewhat redundant regex for detecting `;;` in imports.

Reply:
Also I want to delay adding the OneStatementPerLine check. There are a lot of lines that violate that check, and the intention was to prevent extra semicolons in imports. So I only want to add checks for that specific case.

Reply:
Turns out **NoWhitespaceBefore** was the difficult one. I added the OneStatementPerLine check as well.

Reply:
Sorry, I didn't suppose that it will require more changes. 

Looks like it was a bug in Checkstyle for OneStatementPerLine and it was fixed in 6.9 release [1]. So after updating checkstyle, custom RegexpSingleline may be deleted.

[1] http://checkstyle.sourceforge.net/releasenotes.html#Release_6.9